cmk/special_agents/agent_prism.py

```diff
@@ -55,117 +40,120 @@ def parse_arguments(argv: Sequence[str] | None) -> Args:
-        writer.writerow(
-            [
-                entity["name"],
-                entity["usageStats"]["storage.user_usage_bytes"],
-                entity["usageStats"]["storage.user_capacity_bytes"],
-            ]
-        )
-
+    print(obj)
 
-def output_alerts(requester: Requester) -> Generator[SectionLine, None, None]:
-    needed_context_keys = {"vm_type"}
 
+def output_alerts(requester: Requester) -> None:
     LOGGING.debug("do request..")
     obj = requester.get(
         "alerts",
         parameters={"resolved": "false", "acknowledged": "false"},
     )
     LOGGING.debug("got %d alerts", len(obj["entities"]))
+    write_title("alerts")
+    print(obj)
 
-    yield ("timestamp", "severity", "message", "context")
 
-    for entity in obj["entities"]:
-        # The message is stored as a pattern with placeholders, the
-        # actual values are stored in context_values, the keys in
-        # context_types
-        full_context = dict(zip(entity["contextTypes"], entity["contextValues"]))
-
-        # create a thinned out context we can provide together with the alert data in order to
-        # provide more sophisticated checks in the future (this could be made a cli option, too)
-        thin_context = {k: v for k, v in full_context.items() if k in needed_context_keys}
-
-        # We have seen informational messages in format:
-        # {dev_type} drive {dev_name} on host {ip_address} has the following problems: {err_msg}
-        # In this case the keys have no values so we can not assign it to the message
-        # To handle this, we output a message without assigning the keys
-        try:
-            message = entity["message"].format(**full_context)
-        except KeyError:
-            message = entity["message"]
-
-        # message can contain line breaks which confuses the parser.
-        message = message.replace("\n", r"\n")
-        yield (entity["createdTimeStampInUsecs"], entity["severity"], message, thin_context)
-
-
-# TODO: get rid of CSV and write JSON
 def output_cluster(requester: Requester) -> None:
     LOGGING.debug("do request..")
     obj = requester.get("cluster")
     LOGGING.debug("got %d keys", len(obj.keys()))
 
     write_title("info")
-    writer = gen_csv_writer()
-    writer.writerow(["name", "version"])
-    writer.writerow([obj["name"], obj["version"]])
+    print(obj)
 
 
-# TODO: get rid of CSV and write JSON
 def output_storage_pools(requester: Requester) -> None:
     LOGGING.debug("do request..")
     obj = requester.get("storage_pools")
     LOGGING.debug("got %d entities", len(obj["entities"]))
 
     write_title("storage_pools")
-    writer = gen_csv_writer()
-    writer.writerow(["name", "usage", "capacity"])
+    print(obj)
+
+
+def output_vms(requester: Requester) -> None:
+    write_title("vms")
+    obj = requester.get("vms")
+    print(obj)
+    for element in obj.get("entities"):
+        print("<<<<%s>>>>" % element.get("vmName"))
```

Comment:
we have a piggyback section writer to handle this.
```python
with ConditionalPiggyBackSection(elements["vmName"]):
    print(element)
```
It also takes care of closing the piggy back section. The whole function would be

```python
with SectionWriter("vms") as w:
    obj = requester.get("cms")
    w.append_json(obj)
    with ConditionalPiggyBackSection(element.get("vmName")):
        print(element)
```